File: Reverse-Vowels-of-a-String.js
/**
 * @param {string} s
 * @return {string}
 */

function isVowel(char) {
    const vowels = 'aeiouAEIOU';
    return vowels.includes(char);
}

var reverseVowels = function(s) {
   let i = 0, j = s.length - 1;
    let strArray = s.split(''); 

    while (i < j) {
        while (i < s.length && !isVowel(strArray[i])) {
            i++;
        }
        while (j >= 0 && !isVowel(strArray[j])) {
            j--;
        }

        if (i < j) {
            let temp = strArray[i];
            strArray[i] = strArray[j];
            strArray[j] = temp;
            i++;
            j--;
        }
    }
    return strArray.join('');
};

File: Reverse-Vowels-of-a-String.cs
public class Solution {
    public string ReverseVowels(string s) {
        HashSet<char> vowels = new HashSet<char> { 'a', 'e', 'i', 'o', 'u' }; 
        int i = 0, j = s.Length - 1;
        char[] charArray = s.ToCharArray(); 

        while (i < j) {
            while (i < s.Length && !vowels.Contains(char.ToLower(charArray[i]))) {
                i++;
            }
            while (j >= 0 && !vowels.Contains(char.ToLower(charArray[j]))) {
                j--;
            }

            if (i < j) {
                char temp = charArray[i];
                charArray[i] = charArray[j];
                charArray[j] = temp;
                i++;
                j--;
            }
        }

        return new string(charArray);
    }
}
